openapi.yaml

```diff
@@ -25,254 +37,677 @@ security:
+      tags:
+        - pizza
+        - dough
+      parameters:
+        - in: path
+          name: pizzaID
+          required: true
+          type: string
+          description: ID of a pizza to retrieve
+      responses:
+        200:
+          description: "A single dough recipe"
+          schema:
+            $ref: "#/definitions/DoughResult"
+        404:
+          $ref: "#/responses/404"
+        500:
+          $ref: "#/responses/500"
+  /pizzas/{pizzaID}/dough/{doughID}:
+    patch:
+      description: set the dough for the pizza with ID pizzaID to the dough with ID doughID
+      tags:
+        - pizza
+        - dough
+      parameters:
+        - in: path
+          name: doughID
+          required: true
+          type: string
+          description: "ID of the dough to use"
+        - in: path
+          name: pizzaID
+          required: true
+          type: string
+          description: "ID of a pizza to update"
+      responses:
+        200:
+          description: "Returns true is the resource was updated"
+          schema:
+            type: boolean
+            example: true
+        404:
+          $ref: "#/responses/404"
+        500:
+          $ref: "#/responses/500"
+  /doughs:
+    get:
+      description: "Get all dough recipes"
+      tags:
+        - dough
+      parameters:
+        - in: query
+          name: name
+          required: false
+          type: string
+          description: Search doughs by name
+        - in: query
+          name: gramsFlour
+          required: false
+          type: integer
+          description: Search doughs by grams flour used
+        - in: query
+          name: gramsWater
+          required: false
+          type: integer
+          description: Search doughs by grams water used
+        - in: query
+          name: gramsSugar
+          required: false
+          description: Search doughs by grams sugar used
+          type: integer
+        - in: query
+          name: gramsOliveOil
+          required: false
+          description: Search doughs by grams olive oil used
+          type: integer
+        - in: query
+          name: gramsYeast
+          required: false
+          type: integer
+          description: Search doughs by grams yeast used
+        - in: query
+          name: gramsSalt
+          required: false
+          type: integer
+          description: Search doughs by grams salt used
+        - in: query
+          name: bulkFermentTime
+          required: false
+          type: integer
+          description: Search doughs by bulk ferment time
+        - in: query
+          name: proofTime
+          required: false
+          type: integer
+          description: Search doughs by proof time
+        - in: query
+          name: waterTemp
+          required: false
+          type: integer
+          description: Search doughs by water temperature at mix time
+        - in: query
+          name: flourType
```

Comment:
How many possible flour types will it be? Should this be an `enum` parameter?

Reply:
My goal here was to allow people to be fairly granular in their specifications about flour types, so an enum might be too restrictive. I think the ideal method here would be to have another `flour` resource but that would add a lot of additional complexity to the API.